edpm_ansible/ansible_plugins/modules/container_config_hash.py

```diff
@@ -209,29 +233,20 @@ def _update_hashes(self):
                 continue
             startup_config_json = json.loads(self._slurp(config))
             config_volumes = self._match_config_volumes(startup_config_json)
-            config_hashes = [
-                self._get_config_hash(vol_path) for vol_path in config_volumes
-            ]
-            config_hashes = filter(None, config_hashes)
-            if 'environment' in startup_config_json:
-                old_config_hash = startup_config_json['environment'].get(
-                    'EDPM_CONFIG_HASH', '')
-            if config_hashes is not None and config_hashes:
-                config_hash = '-'.join(config_hashes)
-                if config_hash == old_config_hash:
+            if config_volumes:
+                old_config_hash = startup_config_json.get('edpm_config_hash', '')
+                exclude_from_hash = startup_config_json.get('edpm_exclude_from_hash', [])
+                new_hashes = [
+                    self._set_config_hash(vol_path, exclude_from_hash) for vol_path in config_volumes
+                ]
+                new_hash = '-'.join(new_hashes)
+                if new_hash != old_config_hash:
+                    startup_config_json['edpm_config_hash'] = (new_hash)
```

Comment:
the calculated hash should be part of environment so that the underlying edpm_container_manage and podman module cna detect the different hash.

https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/edpm_ansible/ansible_plugins/modules/edpm_container_manage.py#L299-L300

You have to update tripleo_container_manage if you get the result out of the environment hash.

Reply:
we talked about it in chat, i misunderstood your original comment. I restore `EDPM_CONFIG_HASH` as `environment` variable.